Userland/Libraries/LibWeb/Animations/KeyframeEffect.h

```diff
@@ -97,7 +97,7 @@ class KeyframeEffect : public AnimationEffect {
     Bindings::CompositeOperation composite() const { return m_composite; }
     void set_composite(Bindings::CompositeOperation value) { m_composite = value; }
 
-    WebIDL::ExceptionOr<Vector<JS::Object*>> get_keyframes();
+    WebIDL::ExceptionOr<Vector<JS::NonnullGCPtr<JS::Object>>> get_keyframes();
```

Comment:
This is an unsafe return type (see #23852)
`AK::Vector` uses `malloc()` to allocate its underlying buffer, and the LibJS GC does not scan all malloc memory for live object pointers.

There are two special vector types in LibJS that ensure their underlying buffers get scanned by GC:
- `JS::MarkedVector<T>`: Each `T` in the `MarkedVector` is treated as a strong root.
- `JS::ConservativeVector<T>`: Every pointer-sized value in the `ConservativeVector` is treated as a potential strong root. This is the same behavior as we have for the C++ stack, where anything that looks like a pointer is treated like one for liveness purposes.

In this case, using `JS::MarkedVector` would be ideal.

Reply:
Just for that function, correct? Not for the actual class member?

Reply:
Right! Since we are visiting the vector elements in `visit_edges()`, it's fine to use it as a class member :)
However, if we return the same kind of vector, there will be nobody to visit it!